sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net6.0.cs

```diff
@@ -111,85 +111,79 @@ namespace Azure.Core.Dynamic
     public static partial class BinaryDataExtensions
     {
         public static dynamic ToDynamic(this System.BinaryData data) { throw null; }
-        public static dynamic ToDynamic(this System.BinaryData data, Azure.Core.Dynamic.DynamicJsonNameMapping propertyNameCasing) { throw null; }
-        public static dynamic ToDynamic(this System.BinaryData data, Azure.Core.Dynamic.DynamicJsonOptions options) { throw null; }
-    }
-    public abstract partial class DynamicData
-    {
-        protected DynamicData() { }
-        internal abstract void WriteTo(System.IO.Stream stream);
-        public static void WriteTo(System.IO.Stream stream, Azure.Core.Dynamic.DynamicData data) { }
+        public static dynamic ToDynamic(this System.BinaryData data, Azure.Core.Dynamic.DynamicDataNameMapping propertyNameCasing) { throw null; }
```

Comment:
I wonder if these methods and the enum (parameter) are not JSON specific, i.e. maybe we want to call then ToDynamicFromJson and the parameter being as it was?

Reply:
In practice, they probably only apply to json but nothing about the API would restrict it to this. Is the thought that the more specific naming would help with discoverability?

Reply:
We can definitely be more explicit about this.

Supporting different options based on the data type (e.g. JSON, XML) makes sense - and I guess we don't need to substantially refactor today if the base class only supports JSON.  It will mean some options functionality exists only in subtypes and some in the base class - that makes sense.  The casing piece seems general across anything that has name/value properties - I guess we then copy those across options types and respect them where relevant, without making the assumption that all possible future formats will have the need for this option.

Reply:
If we think that the enum will scale to other formats, then we should keep the name generic. I was just observing that the rename to the main class does not necessitate renames to the extension methods and their parameters. These are independent.

Reply:
@annelo-msft, we would add a parameter to the method to select the format.

Reply:
Updated.  Per this discussion, we have the option to keep a ToDynamic() method with a parameter that specifies the format, but this can be added at a later time.